include/cache.php

```diff
@@ -46,43 +50,71 @@ public function switch_blog( $new_blog_id ) {
 	}
 
 	/**
-	 * Add a value in cache.
+	 * Adds a value in cache.
 	 *
 	 * @since 1.7
+	 * @since 3.6 Returns the cached value.
 	 *
 	 * @param string $key  Cache key.
 	 * @param mixed  $data The value to add to the cache.
-	 * @return void
+	 * @return mixed
+	 *
+	 * @phpstan-param non-empty-string $key
+	 * @phpstan-param TCacheData $data
+	 * @phpstan-return TCacheData
 	 */
-	public function set( $key, $data ) {
+	public function set( string $key, $data ) {
 		$this->cache[ $this->blog_id ][ $key ] = $data;
+		return $data;
 	}
 
 	/**
-	 * Get value from cache.
+	 * Returns value from cache.
 	 *
 	 * @since 1.7
 	 *
 	 * @param string $key Cache key.
 	 * @return mixed
+	 *
+	 * @phpstan-param non-empty-string $key
+	 * @phpstan-return TCacheData|false
 	 */
-	public function get( $key ) {
+	public function get( string $key ) {
 		return isset( $this->cache[ $this->blog_id ][ $key ] ) ? $this->cache[ $this->blog_id ][ $key ] : false;
 	}
 
 	/**
-	 * Clean the cache (for this blog only).
+	 * Cleans the cache (for this blog only).
 	 *
 	 * @since 1.7
 	 *
-	 * @param string $key Cache key.
+	 * @param string $key Optional. Cache key. An empty string to clean the whole cache for the current blog.
+	 *                    Default is an empty string.
 	 * @return void
 	 */
-	public function clean( $key = '' ) {
-		if ( empty( $key ) ) {
+	public function clean( string $key = '' ) {
```

Comment:
I propose not to typehint the existing methods `set()`, `get()`, `clean()`. Although documented as string, nothing prevented to use any possible array key.

Reply:
Done in [09ae96e](https://github.com/polylang/polylang/pull/1403/commits/09ae96eff2f7141c85d9852935db7bc462ddc9e7).